### ISSUE TITLE
Reduce generates invalid c# with dangling semicolon

### ISSUE THREAD

**Issue by bradmarder** [jakobbotsch/Fuzzlyn#15]
Here is a shortened snippet.  On an unrelated note, any plans to upgrade to net7?

```c#
// Generated by Fuzzlyn v1.5 on 2023-01-11 04:06:36
// Run on X64 Windows
// Seed: 4088792501311264625
// Reduced from 212.8 KiB to 2.1 KiB in 00:02:36
// Debug: Outputs -105
// Release: Outputs 105
public interface I0
{
    ulong M4()
    {
        return default(ulong);
    }

    ;
}
```

**Comment by jakobbotsch** [comment]
Thanks for the report. How did you reproduce this problem? On what version of the runtime is `4088792501311264625` a failing example?

> On an unrelated note, any plans to upgrade to net7?

Just did this in 3875e3da19e2fe26beb706b7881ad02c6577c76a, thanks for the reminder!

**Comment by bradmarder** [comment]
> Thanks for the report. How did you reproduce this problem? On what version of the runtime is `4088792501311264625` a failing example?

I am a bit confused with versioning. I'm using a new machine, just downloaded Fuzzlyn and net 7 SDK, compiled it for release mode, and started generating programs. After receiving lots of false-positives (seeds that run with error when targeting net6, but not 7), it occurred to me to the target framework of Fuzzlyn is a factor. Since manually changing the csproj files to net7, I haven't found a single seed yet.

So to answer your question, for this specific seed, I was just passing the net `7.0.2` host, but with Fuzzlyn targeting net6.

**Comment by jakobbotsch** [comment]
If you pass `dotnet.exe` from a shared install location then its default runtime selection will kick in when executing the programs. For your scenario, you would probably want to extract a standalone dotnet of the particular version you want to test (e.g. from the "binaries" section at dot.net). Or similarly, you could download one of the latest builds from https://github.com/dotnet/installer#table. I personally only run Fuzzlyn with my own builds of the runtime and pass corerun as the host.

Note that for .NET versions prior to .NET 7 you also needed to disable `DOTNET_MULTILEVEL_LOOKUP` (some context [here](https://learn.microsoft.com/en-us/dotnet/core/compatibility/deployment/7.0/multilevel-lookup)), but I think that wouldn't be necessary anymore since this is now disabled by default.

Anyway, thanks for adding those extra details. The problem here seems to be that the coarse statement remover incorrectly can add a body to interface declarations:

```diff
@@ -1,6 +1,11 @@
 public interface I0
 {
-    ulong M4();
+    ulong M4()
+    {
+        return default(ulong);
+    }
+
+    ;
     long[] M8()
     {
         return default(long[]);
```

Fixed in 48bcba4.
